pennylane/ops/op_math/parametrized_hamiltonian.py

```diff
@@ -0,0 +1,188 @@
+    r"""Callable object holding the information representing a parametrized Hamiltonian.
+
+    Passing parameters to the ``ParametrizedHamiltonian`` returns an :class:`~pennylane.operation.Operator` representing the Hamiltonian
+    for that set of parameters.
+
+    The Hamiltonian can be represented as a linear combination of other operators, e.g.,
+    :math:`H(v, t) = H_\text{drift} + \sum_j f_j(v, t) H_j`, where the :math:`v` are trainable parameters,
+    and t is time.
+
+    For example, a time-dependent ``ParametrizedHamiltonian`` with a single trainable parameter, :math:`a`, could be :math:`H = 2 X_1 X_2 + \sin(a t) Y_1 Y_2`
+
+    Args:
+        coeffs (Union[float, callable]): coefficients of the Hamiltonian expression, which may be constants or
+            parametrized functions. All functions passed as ``coeffs`` must have the signature ``(params, t)``
+            with ``params`` being trainable parameters and ``t`` being time.
+        observables (Iterable[Observable]): observables in the Hamiltonian expression, of same length as ``coeffs``
+
+    A ``ParametrizedHamiltonian`` is callable with the fixed signature ``H(params, t)`` that returns an
+    ``Operator`` representing an instance of the Hamiltonian with the specified parameter values.
+
+    **Example:**
+
+    .. code-block:: python3
+
+        coeffs = [2, lambda v, t: v[0] * jnp.sin(v[1] * t)]
+        observables =  [qml.PauliX(0), qml.PauliY(1)]
+        H = ParametrizedHamiltonian(coeffs, observables)
+
+    >>> H(jnp.ones(2), 1.)
+    (2*(PauliX(wires=[0]))) + (0.8414710164070129*(PauliY(wires=[1])))
+
+    A ``ParametrizedHamiltonian`` can be created by passing a list of coefficients (scalars or functions), as well as
+    a list of corresponding observables. The functions must all have signatures, ``(params, t)`` with ``params`` being
+    trainable parameters and ``t`` being time.
+
+    >>> def f1(params, t): return np.sin(params[0]*t)
+    >>> def f2(params, t): return params[1] * np.cos(t)
+
+    The functions, along with scalar coefficients, can then be used to initialize a ``ParametrizedHamiltonian``,
+    which will be split into a fixed and parametrized term. The fixed term is an ``Operator``, while the parametrized
+    term must be initialized with concrete parameters to obtain an ``Operator``.
+
+    .. code-block:: python3
+
+        coeffs = [2, f1, f2]
+        obs = [qml.PauliX(0)@qml.PauliX(1), qml.PauliY(0)@qml.PauliY(1), qml.PauliZ(0)@qml.PauliZ(1)]
+        H = ParametrizedHamiltonian(coeffs, obs)
+
+    The resulting object can be passed parameters, and will return an ``Operator`` representing the
+    ``ParametrizedHamiltonian`` with the specified parameters:
+
+    >>> H([1.2, 2.3], 0.5)
+    (2*(PauliX(wires=[0]) @ PauliX(wires=[1]))) + ((0.5646424733950354*(PauliY(wires=[0]) @ PauliY(wires=[1]))) + (2.0184398923478573*(PauliZ(wires=[0]) @ PauliZ(wires=[1]))))
+
+    We can also access the fixed and parametrized terms seperately:
```

Comment:
```suggestion
    
    We can obtain the fixed and parametrized term of the ``ParametrizedHamiltonian``. The fixed term is an ``Operator``, while the parametrized term must be initialized with concrete parameters to obtain an ``Operator``.
```
moved from above making the extra sentence redundant